framework/src/meshgenerators/FancyExtruderGenerator.C

```diff
@@ -510,6 +577,33 @@ FancyExtruderGenerator::generate()
 
         new_elem = mesh->add_elem(new_elem);
 
+        // maintain extra integers
+        if (num_extra_elem_integers > 0)
+        {
+          for (unsigned int i = 0; i < num_extra_elem_integers; i++)
+          {
+            // declare the "control_drum_id" when the first elem is added
+            if (!mesh->has_elem_integer(id_names[i]))
+              mesh->add_elem_integer(id_names[i]);
+            new_elem->set_extra_integer(i, elem->get_extra_integer(i));
+          }
+        }
+
+        if (_elem_integers_swap_pairs.size())
+        {
+          for (unsigned int i = 0; i < _elem_integer_indices_to_swap.size(); i++)
+          {
+            auto & elevation_extra_swap_pairs = _elem_integers_swap_pairs[i * _heights.size() + e];
+
+            auto new_extra_id_it = elevation_extra_swap_pairs.find(
+                elem->get_extra_integer(_elem_integer_indices_to_swap[i]));
+
+            if (new_extra_id_it != elevation_extra_swap_pairs.end())
+              new_elem->set_extra_integer(_elem_integer_indices_to_swap[i],
+                                          new_extra_id_it->second);
```

Comment:
```suggestion
            const auto new_extra_id = elevation_extra_swap_pairs.at(
                elem->get_extra_integer(_elem_integer_indices_to_swap[i]));

            if (new_extra_id_it != elevation_extra_swap_pairs.end())
              new_elem->set_extra_integer(_elem_integer_indices_to_swap[i],
                                          new_extra_id);
```

In the off case that this doesn't exist, we'll get a more meaningful assertion from c++ (hopefully) vs trying to access the beyond-the-end iterator

Reply:
Thanks, @loganharbour 
We have addressed all your comments except for this one on our local branch.
For this one,  I tried (I think that's what you suggested)
```
const auto new_extra_id = elevation_extra_swap_pairs[elem->get_extra_integer(_elem_integer_indices_to_swap[i])];
new_elem->set_extra_integer(_elem_integer_indices_to_swap[i], new_extra_id);
```
The code then just converts any elem extra integers that are not specified in `elem_integers_swaps` to 0 instead of retaining the original values.
I think the original approach works well as `if (new_extra_id_it != elevation_extra_swap_pairs.end())` prevents any access to beyond-the-end iterator. Any more suggestion?

Reply:
Nope - that works for me. I failed to catch the original logic!

Go ahead and push and we should be good to approve. Just please click "resolve conversation" on the comments that you have resolved; it makes the review process easier.